src/ElasticSearchable.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace ScoutEngines\Elasticsearch;
+
+use Laravel\Scout\Builder as ScoutBuilder;
+use Laravel\Scout\Searchable;
+
+trait ElasticSearchable
```

Comment:
I'm not totally agree with making a new trait for it. I understand the `elasticSearch` method to be able to customize the search but as this is a driver it has to be something that must comply with the current usage of Scout using the `Searchable` trait.

We might need to get around the `searchableWithin` method, for example a public property in the Model makes more sense here and then in the Engine just make a ternary.

I guess we can have the trait but we need to make it the `ElasticsearchEngine` to work out of the box with the `Searchable` trait.